spring-cloud-stream/src/test/java/org/springframework/cloud/stream/binder/HealthIndicatorsConfigurationTests.java

```diff
@@ -73,6 +82,29 @@ public void healthIndicatorsCheck() throws Exception {
 		assertThat(healthIndicators.get("binder2").health().getStatus(), CoreMatchers.equalTo(Status.UNKNOWN));
 	}
 
+	@Test
+	public void healthIndicatorsCheckWhenDisabled() throws Exception {
+		ConfigurableApplicationContext context =
+				createBinderTestContext(
+						new String[]{"binder1", "binder2"}, "spring.cloud.stream.defaultBinder:binder2",
+						"management.health.binders.enabled:false");
+
+		Binder binder1 = context.getBean(BinderFactory.class).getBinder("binder1");
+		assertThat(binder1, instanceOf(StubBinder1.class));
+		Binder binder2 = context.getBean(BinderFactory.class).getBinder("binder2");
+		assertThat(binder2, instanceOf(StubBinder2.class));
+		boolean exceptionThrown = false;
+		try {
+			context.getBean("bindersHealthIndicator", CompositeHealthIndicator.class);
+		}
+		catch (NoSuchBeanDefinitionException e) {
+			exceptionThrown = true;
+		}
+		assertTrue(exceptionThrown);
```

Comment:
We should probably add a message explaining the nature of the failure, or perhaps even swallow the exception but add a fail() in the try clause.